internal/builders/dockerfile/docker/cmd.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2022 SLSA Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docker
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// CommandSteps is a series of steps executed when running a build. These steps
+// can be included in the buildConfig in provenance statements.
+type CommandSteps []*CommandStep
+
+// Run executes each step. If any step returns an error the error is returned
+// and subsequent steps are not executed.
+func (s CommandSteps) Run() error {
+	for _, step := range s {
+		if err := step.Run(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// CommandStep is a command executed by the builder.
+type CommandStep struct {
```

Comment:
I'm thinking of making this kind of a standard thing in the library that all the builders that run commands can use in their build config.

That way it makes the build config's similar and a bit more standardized.